cardano-lib/mainnet-config.nix

```diff
@@ -6,7 +6,10 @@
 {
   ##### Locations #####
 
-  GenesisFile = ./mainnet-byron-genesis.json;
+  ByronGenesisFile = ./mainnet-byron-genesis.json;
+  ByronGenesisHash = "5f20df933584822601f9e3f8c024eb5eb252fe8cefb24d1317dc3d432e940ebb";
+  ShelleyGenesisFile = ./mainnet-shelley-genesis.json;
+  ShelleyGenesisHash = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
```

Comment:
Is this field read? We should use it to check whether the right (unmodified) genesis file is being used.